### PR TITLE
Project Movies – Caroline Birgersson

### PR DESCRIPTION
**Pull request by carolinebirgersson**
Link to my deployed project: https://popular-movies-app.netlify.app/

My reflections on how this week's project turned out: Pretty good, I liked Vans lessons this Monday - it helped to get started with the project. If I had more time I would have added a not found-page + loading. 

Things I like to explain: The return button worked until the end, now it only returns in mobile – and I haven't changed anything in the js – nor the css that should effect the desktop version of the button. So if the code reviewers could solve that I would be grateful :)

### CODE REVIEW

File: code/src/pages/MovieStartPage.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import React, { useState, useEffect } from "react";
+
+import { MovieCard } from "../components/js/MovieCard";
+
+import "..//components/css/moviestartpage.css";
+
+export const MovieStartPage = () => {
+  const API_KEY = "88b00ce6c0c72d59de65e4b1fcce3a85";
```
Comment (ankimelin) [inline code/src/pages/MovieStartPage.js:8]: Nice to have this separated.

File: code/src/pages/MovieStartPage.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  const [movies, setMovie] = useState([]);
+
+  useEffect(() => {
+    const URL_MOVIES = `https://api.themoviedb.org/3/movie/popular?api_key=${API_KEY}&language=en-US&page=1`;
+
+    fetch(URL_MOVIES)
+      .then((response) => response.json())
+      .then((json) => setMovie(json.results));
```
Comment (ankimelin) [inline code/src/pages/MovieStartPage.js:16]: When we only pass one argument to a function (here .then is the function?) we can actually remove the () around that argument, Maks/Van told me during a 1:1 code session. So you could, if you like, `put .then(response => response.json())` and `.then(json => setMovie(json.results))` instead.

File: code/src/pages/MovieStartPage.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  }, []);
+
+  return (
+    <section className="movie-start-page">
+      {movies.map((movie) => (
```
Comment (ankimelin) [inline code/src/pages/MovieStartPage.js:21]: Same here, you can remove the () from movie if you want!

File: code/src/pages/MovieStartPage.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        <MovieCard key={movie.id} {...movie} />
```
Comment (ankimelin) [inline code/src/pages/MovieStartPage.js:22]: So cool with the `{...movie}`!! Superclean!

File: code/src/pages/MovieBio.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+import { BackButton } from "components/js/BackButton";
+
+import "components/css/moviebio.css";
+
+export const MovieBio = () => {
+  const { id } = useParams();
+  const [bio, setBio] = useState({});
+  const API_KEY = "88b00ce6c0c72d59de65e4b1fcce3a85";
+  const history = useHistory();
```
Comment (ankimelin) [inline code/src/pages/MovieBio.js:12]: cool that you got to use history!

File: code/src/pages/MovieBio.js [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+  useEffect(() => {
+    const URL_MOVIE_DETAILS = `https://api.themoviedb.org/3/movie/${id}?api_key=${API_KEY}&language=en-US`;
+    fetch(URL_MOVIE_DETAILS)
+      .then((response) => response.json())
+      .then((json) => {
```
Comment (ankimelin) [inline code/src/pages/MovieBio.js:18]: () could be removed around response and json, see comment below!

File: code/src/pages/MovieBio.js [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+        setBio(json);
+      })
+      .catch((error) => {
+        console.error("Bad request", error);
```
Comment (ankimelin) [inline code/src/pages/MovieBio.js:22]: Really like that you have error handling, well done!

File: code/src/pages/MovieBio.js [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+import { useParams, Link, useHistory } from "react-router-dom";
+
+import { BackButton } from "components/js/BackButton";
+
+import "components/css/moviebio.css";
+
+export const MovieBio = () => {
+  const { id } = useParams();
+  const [bio, setBio] = useState({});
+  const API_KEY = "88b00ce6c0c72d59de65e4b1fcce3a85";
```
Comment (ankimelin) [inline code/src/pages/MovieBio.js:11]: maybe put the api key either above or below the use-thingies? it feels like the odd one out in that group :-)

File: code/src/pages/MovieBio.js [hunk 9]
```diff
@@ -0,0 +1,53 @@
+import React, { useState, useEffect } from "react";
+import { useParams, Link, useHistory } from "react-router-dom";
+
+import { BackButton } from "components/js/BackButton";
```
Comment (ankimelin) [inline code/src/pages/MovieBio.js:4]: Nice to have the back button as a separate component, if I ever implement it I will checkout your approach for this!

File: code/src/index.css [hunk 10]
```diff
@@ -1,13 +1,18 @@
...
+}
+
 body {
   margin: 0;
   font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen",
     "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue",
     sans-serif;
   -webkit-font-smoothing: antialiased;
   -moz-osx-font-smoothing: grayscale;
+  background: black;
```
Comment (ankimelin) [inline code/src/index.css:12]: Is this standard, to put the background to black, or how did you come to think about that?

File: code/src/components/js/MovieCard.js [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+import "..//css/moviecard.css";
+
+export const MovieCard = ({ id, title, release_date, poster_path }) => {
+  return (
+    <div className="movie-card">
+      <Link to={`/movies/${id}`}>
+        <img
+          className="movie-image"
+          src={`https://image.tmdb.org/t/p/w500${poster_path}`}
+          alt={title}
```
Comment (ankimelin) [inline code/src/components/js/MovieCard.js:13]: superclear to have these on separate lines

File: code/src/components/js/BackButton.js [hunk 12]
```diff
@@ -0,0 +1,13 @@
...
+
+export const BackButton = () => {
+  return (
+    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 30 30">
+      <path
+        d="M27 14.5C27 7.596441 21.4035594 2 14.5 2S2 7.596441 2 14.5 7.5964406 27 14.5 27 27 21.403559 27 14.5zm-19.3388348-.353553l7.4852814-7.485282c.1952622-.195262.5118446-.195262.7071068 0l2.1213203 2.121321c.1952622.195262.1952622.511844 0 .707106L12.9644661 14.5l5.0104076 5.010408c.1952622.195262.1952622.511844 0 .707106l-2.1213203 2.121321c-.1952622.195262-.5118446.195262-.7071068 0l-7.4852814-7.485282c-.19799-.19799-.197989-.509117 0-.707106z"
+        fill="#fff"
+        fillRule="evenodd"
+      />
+    </svg>
```
Comment (ankimelin) [inline code/src/components/js/BackButton.js:11]: This looks superadvanced to me, can you explain a little, what is it, how did you know that you can go with this?

File: code/src/components/css/moviecard.css [hunk 13]
```diff
@@ -0,0 +1,70 @@
...
+
+/* Tablet */
+@media (min-width: 651px) and (max-width: 1024px) {
+  .movie-title {
+    font-size: 20px;
+    margin: 0;
+  }
+}
+
+/* Mobile */
```
Comment (ankimelin) [inline code/src/components/css/moviecard.css:53]: Great work to make the site responsive!

File: code/src/components/css/moviecard.css [hunk 14]
```diff
@@ -0,0 +1,70 @@
...
+  display: flex;
+  height: auto;
+  position: relative;
+  width: 25%;
+}
+
+.movie-card:hover .card-content-container {
+  transition: 0.2s ease-in-out;
+  opacity: 1;
+}
```
Comment (ankimelin) [inline code/src/components/css/moviecard.css:11]: Interesting to have a transition on this, why did you go with that?

**Review by ankimelin — APPROVED** [review]
Great work with the Movie project Caroline!! 

Your code is easy to understand - you have kept it clean and readable and it looks like you have put some time to plan the components, making the structure easy to understand aswell. So cool that you made the hover styling workout, and the back button to work! The back button works in Chrome in my computer. 

Furthermore you have the css files separated which also helps readability, and a tough one this week - naming of pages and components - you did very well!

File: code/src/components/css/moviebio.css [hunk 15]
```diff
@@ -0,0 +1,136 @@
+.bio-section {
+  display: flex;
+  color: white;
+  align-items: center;
+  position: relative;
+  z-index: 0;
+}
+
+.background-image {
+  position: relative;
```
Comment (EmOlsz) [inline code/src/components/css/moviebio.css:10]: This property doesn't really do anything. If we use `position: relative`, we usually want to set it children as `position: absolute` or change top/right/bottom/left properties. It's none of your cases, so we can delete it :)

File: code/src/pages/MovieBio.js [hunk 16]
```diff
@@ -0,0 +1,53 @@
...
+      });
+  }, [id]);
+
+  return (
+    <section className="bio-section">
+      <img
+        className="background-image"
+        src={`https://image.tmdb.org/t/p/original${bio.backdrop_path}`}
+        alt={bio.title}
+      ></img>
```
Comment (EmOlsz) [inline code/src/pages/MovieBio.js:32]: `img` is self-closing tag

File: code/src/pages/MovieStartPage.js [hunk 17]
```diff
@@ -0,0 +1,26 @@
+import React, { useState, useEffect } from "react";
+
+import { MovieCard } from "../components/js/MovieCard";
+
+import "..//components/css/moviestartpage.css";
```
Comment (EmOlsz) [inline code/src/pages/MovieStartPage.js:5]: ups, small typo in file path ;)

File: code/src/pages/MovieBio.js [hunk 18]
```diff
@@ -0,0 +1,53 @@
...
+      <Link className="back-button" onClick={() => history.goBack()}>
```
Comment (EmOlsz) [inline code/src/pages/MovieBio.js:33]: Your button is covered by `.content-container` element (being precise, by its `padding`). Needs to be fixed

**Pull request closed without merging**